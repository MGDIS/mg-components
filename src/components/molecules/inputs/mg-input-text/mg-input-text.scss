@use '../styles/mg-input.scss';
@use '../styles/mg-input-box.scss';
@use '../styles/mg-input-width.scss';
@use '../styles/mg-input-charleft.scss';
@use '../../../../styles/a11y.scss';

:host {
  display: block;
}

/**
* Icon in input
*/

.mg-input.mg-input--has-icon .mg-input__input {
  position: relative;
  & > mg-icon {
    // layout
    position: absolute;
    top: 50%;
    left: var(--mg-inputs-spacer);
    // text
    color: var(--mg-inputs-color);
    // other
    transform: translateY(-50%);
  }
  .mg-input__box {
    padding-left: calc(var(--mg-icon-regular-size) + var(--mg-inputs-spacer) * 2);
  }
}
