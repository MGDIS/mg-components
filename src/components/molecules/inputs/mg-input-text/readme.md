# mg-input-text

<!-- Auto Generated Below -->


## Properties

| Property                | Attribute                 | Description                                                                                                                                    | Type      | Default                     |
| ----------------------- | ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- | --------- | --------------------------- |
| `characterLeftTemplate` | `character-left-template` | Template to use for characters left sentence                                                                                                   | `string`  | `undefined`                 |
| `disabled`              | `disabled`                | Define if input is disabled                                                                                                                    | `boolean` | `false`                     |
| `displayCharacterLeft`  | `display-character-left`  | Define if component should display character left                                                                                              | `boolean` | `true`                      |
| `helpText`              | `help-text`               | Template to use for characters left sentence                                                                                                   | `string`  | `undefined`                 |
| `identifier`            | `identifier`              | Identifier is used for the element ID (id is a reserved prop in Stencil.js) If not set, it will be created.                                    | `string`  | `createID('mg-input-text')` |
| `invalid`               | `invalid`                 | Define input pattern error message                                                                                                             | `boolean` | `undefined`                 |
| `label` _(required)_    | `label`                   | Input label Required                                                                                                                           | `string`  | `undefined`                 |
| `labelColon`            | `label-colon`             | Define if label has colon ":"                                                                                                                  | `boolean` | `false`                     |
| `labelOnTop`            | `label-on-top`            | Define if label is displayed on top                                                                                                            | `boolean` | `undefined`                 |
| `maxlength`             | `maxlength`               | Input max length                                                                                                                               | `number`  | `400`                       |
| `name`                  | `name`                    | Input name If not set the value equals the identifier                                                                                          | `string`  | `this.identifier`           |
| `pattern`               | `pattern`                 | Define input pattern to validate                                                                                                               | `string`  | `undefined`                 |
| `patternErrorMessage`   | `pattern-error-message`   | Define input pattern error message                                                                                                             | `string`  | `undefined`                 |
| `placeholder`           | `placeholder`             | Input placeholder. It should be a word or short phrase that demonstrates the expected type of data, not a replacement for labels or help text. | `string`  | `undefined`                 |
| `readonly`              | `readonly`                | Define if input is readonly                                                                                                                    | `boolean` | `false`                     |
| `required`              | `required`                | Define if input is required                                                                                                                    | `boolean` | `false`                     |
| `tooltip`               | `tooltip`                 | Add a tooltip message next to the input                                                                                                        | `string`  | `undefined`                 |
| `valid`                 | `valid`                   | Define input pattern to validate                                                                                                               | `boolean` | `undefined`                 |
| `value`                 | `value`                   | Component value                                                                                                                                | `string`  | `undefined`                 |


## Events

| Event         | Description                     | Type                  |
| ------------- | ------------------------------- | --------------------- |
| `valueChange` | Emmited event when value change | `CustomEvent<string>` |


## Dependencies

### Depends on

- [mg-label](../../../atoms/mg-label)
- [mg-tooltip](../../../atoms/mg-tooltip)
- [mg-icon](../../../atoms/mg-icon)
- [mg-character-left](../../../atoms/mg-character-left)

### Graph
```mermaid
graph TD;
  mg-input-text --> mg-label
  mg-input-text --> mg-tooltip
  mg-input-text --> mg-icon
  mg-input-text --> mg-character-left
  style mg-input-text fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
