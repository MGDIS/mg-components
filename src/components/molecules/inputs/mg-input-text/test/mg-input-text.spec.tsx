import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { MgInputText } from '../mg-input-text';
import { MgButton } from '../../../../atoms/mg-button/mg-button';
import { MgIcon } from '../../../../atoms/mg-icon/mg-icon';
import { messages } from '../../../../../locales';

const getPage = (args, content?) => {
  const page = newSpecPage({
    components: [MgInputText, MgButton, MgIcon],
    template: () => <mg-input-text {...args}>{content}</mg-input-text>,
  });

  jest.runAllTimers();

  return page;
};

describe('mg-input-text', () => {
  beforeEach(() => jest.useFakeTimers());
  afterEach(() => jest.runOnlyPendingTimers());
  test.each([
    { label: 'label', identifier: 'identifier' },
    { label: 'label', identifier: 'identifier', labelHide: true },
    { label: 'label', identifier: 'identifier', labelOnTop: true },
    { label: 'label', identifier: 'identifier', readonly: true },
    { label: 'label', identifier: 'identifier', type: 'search' },
    { label: 'label', identifier: 'identifier', type: 'search', icon: 'magnifying-glass' },
    { label: 'label', identifier: 'identifier', readonly: true, labelOnTop: true, tooltip: 'Tooltip message' },
    { label: 'label', identifier: 'identifier', readonly: true, value: 'blu' },
    { label: 'label', identifier: 'identifier', tooltip: 'My Tooltip Message' },
    { label: 'label', identifier: 'identifier', tooltip: 'My Tooltip Message', labelOnTop: true },
  ])('Should render with args %s:', async args => {
    const { root } = await getPage(args);
    expect(root).toMatchSnapshot();
  });

  test.each([
    <mg-button slot="append-input" label="search" identifier="identifier">
      <mg-icon icon="magnifying-glass"></mg-icon> Search
    </mg-button>,
    <span slot="append-input">test</span>,
  ])('Should render with slot.', async slot => {
    const args = { label: 'label', identifier: 'identifier', type: 'search' };

    const page = await getPage(args, slot);
    expect(page.root).toMatchSnapshot();
  });

  test.each(['', undefined])('Should throw error with invalid label property : %s', async value => {
    try {
      await getPage({ label: value });
    } catch (err) {
      expect(err.message).toMatch('prop "label" is required');
    }
  });

  test('Should throw an error with labelOnTop & labelHide set to true', async () => {
    try {
      await getPage({ label: 'batman', labelOnTop: true, labelHide: true });
    } catch (err) {
      expect(err.message).toMatch('<mg-input> prop "labelOnTop" must not be paired with the prop "labelHide"');
    }
  });

  test.each(['', undefined])('Should throw an error when pattern is used with patternErrorMessage: %s', async value => {
    try {
      const { root } = await getPage({ label: 'blu', pattern: '[a-z]*', patternErrorMessage: value });
      expect(root).toMatchSnapshot();
    } catch (err) {
      expect(err.message).toMatch('<mg-input-text> prop "pattern" must be paired with the prop "patternErrorMessage"');
    }
  });

  test('Should trigger events', async () => {
    const inputValue = 'Blu';
    const args = { label: 'label', identifier: 'identifier', helpText: 'My help text' };
    const page = await getPage(args);

    const element = page.doc.querySelector('mg-input-text');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(() => true);
    Object.defineProperty(input, 'validity', {
      get: jest.fn(() => ({
        valueMissing: false,
        patternMismatch: false,
      })),
    });

    jest.spyOn(page.rootInstance.valueChange, 'emit');

    input.dispatchEvent(new CustomEvent('focus', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.classList.has('is-focused')).toBeTruthy();

    expect(page.root).toMatchSnapshot(); //Snapshot on focus

    input.value = inputValue;
    input.dispatchEvent(new CustomEvent('input', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.valueChange.emit).toHaveBeenCalledWith(inputValue);

    input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
    await page.waitForChanges();
    expect(page.rootInstance.classList.has('is-focused')).toBeFalsy();
  });

  test.each([
    { validity: true, valueMissing: false, patternMismatch: false },
    { validity: false, valueMissing: true, patternMismatch: false },
    { validity: false, valueMissing: false, patternMismatch: true },
  ])('validity (%s), valueMissing (%s), patternMismatch (%s)', async ({ validity, valueMissing, patternMismatch }) => {
    const args = { label: 'label', identifier: 'identifier', patternErrorMessage: 'Non' };
    const page = await getPage(args);

    const element = page.doc.querySelector('mg-input-text');
    const input = element.shadowRoot.querySelector('input');

    //mock validity
    input.checkValidity = jest.fn(() => validity);
    Object.defineProperty(input, 'validity', {
      get: jest.fn(() => ({
        valueMissing,
        patternMismatch,
      })),
    });

    input.dispatchEvent(new CustomEvent('blur', { bubbles: true }));
    await page.waitForChanges();

    if (validity) {
      expect(page.rootInstance.errorMessage).toBeUndefined();
    } else if (valueMissing) {
      expect(page.rootInstance.errorMessage).toEqual(messages.errors.required);
    } else if (patternMismatch) {
      expect(page.rootInstance.errorMessage).toEqual(args.patternErrorMessage);
    }
    expect(page.rootInstance.valid).toEqual(validity);
    expect(page.rootInstance.invalid).toEqual(!validity);
  });
});
