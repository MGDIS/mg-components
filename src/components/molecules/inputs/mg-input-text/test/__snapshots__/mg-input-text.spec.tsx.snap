// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mg-input-text Should render with args {
  label: 'label',
  identifier: 'identifier',
  displayCharacterLeft: false
}: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 0;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args {
  label: 'label',
  identifier: 'identifier',
  readonly: true,
  labelOnTop: true,
  tooltip: 'Tooltip message'
}: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--label-on-top mg-input--readonly mg-input--text mg-input--width-full">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
      </div>
      <div class="mg-input__input-container">
        <strong></strong>
        <slot name="append-input"></slot>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args {
  label: 'label',
  identifier: 'identifier',
  readonly: true,
  value: 'blu'
}: 1`] = `
<mg-input-text value="blu">
  <mock:shadow-root>
    <div class="mg-input mg-input--readonly mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong>
          blu
        </strong>
        <slot name="append-input"></slot>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args {
  label: 'label',
  identifier: 'identifier',
  tooltip: 'My Tooltip Message'
}: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
          <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
            <mg-icon>
              <mock:shadow-root>
                <svg aria-hidden="true" class="mg-icon mg-icon--info-circle mg-icon--size-regular" focusable="false" viewBox="0 0 16 16">
                  <path d="M8 .67A7.33 7.33 0 1 0 15.33 8 7.33 7.33 0 0 0 8 .67ZM9 12a.29.29 0 0 1-.3.3H7.35a.29.29 0 0 1-.3-.3V7.32A.29.29 0 0 1 7.33 7h1.32a.29.29 0 0 1 .35.3Zm0-6.74a.29.29 0 0 1-.3.3H7.35a.29.29 0 0 1-.3-.3V4a.29.29 0 0 1 .28-.3h1.32a.29.29 0 0 1 .3.28V4Z" fill="currentColor"></path>
                </svg>
              </mock:shadow-root>
            </mg-icon>
          </mg-tooltip>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args {
  label: 'label',
  identifier: 'identifier',
  tooltip: 'My Tooltip Message',
  labelOnTop: true
}: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--label-on-top mg-input--text mg-input--width-full">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
        <mg-tooltip identifier="identifier-tooltip" message="My Tooltip Message">
          <mg-icon>
            <mock:shadow-root>
              <svg aria-hidden="true" class="mg-icon mg-icon--info-circle mg-icon--size-regular" focusable="false" viewBox="0 0 16 16">
                <path d="M8 .67A7.33 7.33 0 1 0 15.33 8 7.33 7.33 0 0 0 8 .67ZM9 12a.29.29 0 0 1-.3.3H7.35a.29.29 0 0 1-.3-.3V7.32A.29.29 0 0 1 7.33 7h1.32a.29.29 0 0 1 .35.3Zm0-6.74a.29.29 0 0 1-.3.3H7.35a.29.29 0 0 1-.3-.3V4a.29.29 0 0 1 .28-.3h1.32a.29.29 0 0 1 .3.28V4Z" fill="currentColor"></path>
              </svg>
            </mock:shadow-root>
          </mg-icon>
        </mg-tooltip>
      </div>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args {
  label: 'label',
  identifier: 'identifier',
  type: 'search',
  icon: 'magnifying-glass'
}: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--has-icon mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <mg-icon>
              <mock:shadow-root>
                <svg aria-hidden="true" class="mg-icon mg-icon--magnifying-glass mg-icon--size-regular" focusable="false" viewBox="0 0 16 16">
                  <path d="M11.24 9.5c1.48-2.44.71-5.62-1.73-7.1-.05-.03-.11-.07-.17-.1A5.232 5.232 0 0 0 2.52 4C.94 6.38 1.6 9.59 3.98 11.17a5.142 5.142 0 0 0 5.54.1L12.3 14c.39.39 1.02.39 1.41 0l.33-.33a.996.996 0 0 0 0-1.41l-2.8-2.76Zm-4.4.5c-1.76.02-3.2-1.4-3.22-3.16-.02-1.76 1.4-3.2 3.16-3.22 1.76-.02 3.2 1.4 3.22 3.16v.06A3.193 3.193 0 0 1 6.84 10Z" fill="currentColor"></path>
                </svg>
              </mock:shadow-root>
            </mg-icon>
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="search">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args { label: 'label', identifier: 'identifier' }: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args { label: 'label', identifier: 'identifier', labelHide: true }: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title class="sr-only" identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args { label: 'label', identifier: 'identifier', labelOnTop: true }: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--label-on-top mg-input--text mg-input--width-full">
      <div class="mg-input__title">
        <mg-input-title identifier="identifier">
          label
        </mg-input-title>
      </div>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args { label: 'label', identifier: 'identifier', readonly: true }: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--readonly mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
        <slot name="append-input"></slot>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with args { label: 'label', identifier: 'identifier', type: 'search' }: 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="search">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text Should render with slot. 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--is-input-group-append mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="search">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <mg-button slot="append-input">
    <!---->
    <button aria-disabled="false" aria-label="search" class="mg-button mg-button--primary" id="identifier" type="submit">
      <div class="mg-button__content">
        <mg-icon>
          <mock:shadow-root>
            <svg aria-hidden="true" class="mg-icon mg-icon--magnifying-glass mg-icon--size-regular" focusable="false" viewBox="0 0 16 16">
              <path d="M11.24 9.5c1.48-2.44.71-5.62-1.73-7.1-.05-.03-.11-.07-.17-.1A5.232 5.232 0 0 0 2.52 4C.94 6.38 1.6 9.59 3.98 11.17a5.142 5.142 0 0 0 5.54.1L12.3 14c.39.39 1.02.39 1.41 0l.33-.33a.996.996 0 0 0 0-1.41l-2.8-2.76Zm-4.4.5c-1.76.02-3.2-1.4-3.22-3.16-.02-1.76 1.4-3.2 3.16-3.22 1.76-.02 3.2 1.4 3.22 3.16v.06A3.193 3.193 0 0 1 6.84 10Z" fill="currentColor"></path>
            </svg>
          </mock:shadow-root>
        </mg-icon>
        Search
      </div>
    </button>
  </mg-button>
</mg-input-text>
`;

exports[`mg-input-text Should render with slot. 2`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--is-append-input-slot-content mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="search">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <span slot="append-input">
    test
  </span>
</mg-input-text>
`;

exports[`mg-input-text Should render with slot. 3`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--has-buttons-group-append mg-input--is-input-group-append mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="search">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <mg-button is-icon="" slot="append-input">
    <!---->
    <button aria-disabled="false" aria-label="cancel" class="mg-button mg-button--icon mg-button--primary" id="identifier-cancel" type="submit">
      <div class="mg-button__content">
        <mg-icon>
          <mock:shadow-root>
            <svg aria-hidden="true" class="mg-icon mg-icon--cross mg-icon--size-regular" focusable="false" viewBox="0 0 16 16">
              <path d="M14.26 12.74 9.52 8l4.7-4.7a.29.29 0 0 0 0-.41l-1.1-1.1a.29.29 0 0 0-.41 0L8 6.48 3.26 1.74a.24.24 0 0 0-.34 0L1.74 2.91a.24.24 0 0 0 0 .34L6.48 8l-4.7 4.7a.29.29 0 0 0 0 .41l1.1 1.1a.29.29 0 0 0 .41 0L8 9.52l4.74 4.74a.24.24 0 0 0 .34 0l1.17-1.17a.24.24 0 0 0 0-.34Z" fill="currentColor"></path>
            </svg>
          </mock:shadow-root>
        </mg-icon>
      </div>
    </button>
  </mg-button>
  <mg-button is-icon="" slot="append-input">
    <!---->
    <button aria-disabled="false" aria-label="validate" class="mg-button mg-button--icon mg-button--primary" id="identifier-validate" type="submit">
      <div class="mg-button__content">
        <mg-icon>
          <mock:shadow-root>
            <svg aria-hidden="true" class="mg-icon mg-icon--check mg-icon--size-regular" focusable="false" viewBox="0 0 16 16">
              <path d="m15.27 4.23-1.11-1.18c-.11-.1-.27-.1-.38 0L6 10.36l-3.76-4a.187.187 0 0 0-.26 0L.71 7.6c-.07.07-.07.19 0 .26l5.12 5.41c.07.07.18.07.25 0l1.3-1.23 7.87-7.45c.1-.1.11-.26.01-.37h.01Z" fill="currentColor"></path>
            </svg>
          </mock:shadow-root>
        </mg-icon>
      </div>
    </button>
  </mg-button>
</mg-input-text>
`;

exports[`mg-input-text Should trigger events 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="is-focused mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-help-text" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
        <div class="mg-input__help-text" id="identifier-help-text">
          My help text
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text display error message with locale: fr 1`] = `
<mg-input-text lang="fr">
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input mg-input__input--has-error">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-error" class="mg-input__box" id="identifier" maxlength="400" name="identifier" required="" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          Ce champ est obligatoire.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text display error message with locale: xx 1`] = `
<mg-input-text lang="xx">
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input mg-input__input--has-error">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-error" class="mg-input__box" id="identifier" maxlength="400" name="identifier" required="" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text display error with displayError component's public method 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left" class="mg-input__box" id="identifier" maxlength="400" name="identifier" required="" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text display error with displayError component's public method 2`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier" required="">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input mg-input__input--has-error">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-error" class="mg-input__box" id="identifier" maxlength="400" name="identifier" required="" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text validity, case next state is disabled validity ({ validity: false, valueMissing: true, patternMismatch: false }), valueMissing (%s), patternMismatch (%s) 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input mg-input__input--has-error">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-error" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text validity, case next state is disabled validity ({ validity: false, valueMissing: true, patternMismatch: false }), valueMissing (%s), patternMismatch (%s) 2`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input mg-input__input--has-error">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-error" class="mg-input__box" disabled="" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text validity, case next state is readonly validity ({ validity: false, valueMissing: true, patternMismatch: false }), valueMissing (%s), patternMismatch (%s) 1`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <div class="mg-input__input mg-input__input--has-error">
          <div class="mg-input__with-character-left" style="--mg-character-left-message-length: 7;">
            <input aria-describedby="identifier-character-left identifier-error" class="mg-input__box" id="identifier" maxlength="400" name="identifier" type="text">
            <mg-character-left identifier="identifier-character-left" maxlength="400"></mg-character-left>
          </div>
          <slot name="append-input"></slot>
        </div>
        <div aria-live="assertive" class="mg-input__error" id="identifier-error">
          This field is required.
        </div>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;

exports[`mg-input-text validity, case next state is readonly validity ({ validity: false, valueMissing: true, patternMismatch: false }), valueMissing (%s), patternMismatch (%s) 2`] = `
<mg-input-text>
  <mock:shadow-root>
    <div class="mg-input mg-input--readonly mg-input--text mg-input--width-full">
      <mg-input-title identifier="identifier">
        label
      </mg-input-title>
      <div class="mg-input__input-container">
        <strong></strong>
        <slot name="append-input"></slot>
      </div>
    </div>
  </mock:shadow-root>
</mg-input-text>
`;
